wlog/logentry.go

```diff
@@ -48,6 +48,7 @@ type LeveledLogger interface {
 	Warn(msg string, params ...Param)
 	Error(msg string, params ...Param)
 	SetLevel(level LogLevel)
+	Enabled(level LogLevel) bool
```

Comment:
My only hesitation here is that adding a new function to an interface is a breaking change for the interface. However, I see that this is required because the wrapped logger wraps a `LeveledLogger` and that's the level where you want to avoid the param conversion.

If we really wanted to keep API stability, the proper path would probably be to do something like defining an extension of the `LeveledLogger` interface that adds `Enabled` and then updating the wrapped logger implementation such that it makes use of the extra `Enabled` functionality only if it exists -- this would maintain interface compatibility both for implementors and for users of wrapped loggers.

However, I do believe that most (if not all) implementations of this interface are within this same module, in which case the "break" is less of an issue since there's unlikely to be module incompatibility.

So, options are basically:

1. Introduce the break but release normally (no-major version) -- live with the fact that although this is technically an API break, we don't expect it to impact anything outside the module so it should be fine
2. Change the implementation in this PR to make it strictly API-additive such that no breaks occur
3. Introduce the break and release with a new major version (don't think this path is worth it)

@bmoylan @tabboud do you guys have any thoughts? I think I'm okay with 1, but want to make sure that we explicitly acknowledge it and are on the same page.

Reply:
Agreed! Went ahead with a different approach which introduces an optional `LevelProvider` interface.